chia/wallet/wallet_node.py

```diff
@@ -822,8 +843,65 @@ def validate_removals(self, coins, proofs, root):
                         return False
         return True
 
-    async def get_additions(self, peer: WSChiaConnection, block_i, additions) -> Optional[List[Coin]]:
-        if len(additions) > 0:
+    async def fetch_puzzle_solution(self, peer, height: uint32, coin: Coin) -> CoinSolution:
+        solution_response = await peer.request_puzzle_solution(
+            wallet_protocol.RequestPuzzleSolution(coin.name(), height)
+        )
+        if solution_response is None or not isinstance(solution_response, wallet_protocol.RespondPuzzleSolution):
```

Comment:
I think I've seen this before in the codebase: The None-check here is redundant.